sdk/python/kfp_tekton/compiler/compiler.py

```diff
@@ -388,6 +388,21 @@ def _create_pipeline_workflow(self, args, pipeline, op_transformers=None, pipeli
       if pipeline_conf.timeout:
         pipelinerun['spec']['timeout'] = '%ds' % pipeline_conf.timeout
 
+      # generate the Tekton service account template
+      service_template = {}
+      if len(pipeline_conf.image_pull_secrets) > 0:
+        service_template = {
+          'apiVersion': 'v1',
+          'kind': 'ServiceAccount',
+          'metadata': {'name': 'secrets'}
```

Comment:
Actually can we make the generated service account to be based on the pipelinerun name?

Reply:
sure